Content.Shared/Species/Components/ReformComponent.cs

```diff
@@ -0,0 +1,48 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Species.Components;
+
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class ReformComponent : Component
+{
+    /// <summary>
+    /// The action to use.
+    /// </summary>
+    [DataField(required: true)]
+    public EntProtoId? ActionPrototype;
+
+    [DataField, AutoNetworkedField] 
+    public EntityUid? ActionEntity;
+
+    /// <summary>
+    /// How long it will take to reform
+    /// </summary>
+    [DataField("reformTime", required: true)]
+    public float ReformTime = 0;
+
+    /// <summary>
+    /// Whether or not the entity should start with a cooldown
+    /// </summary>
+    [DataField("startDelayed")]
+    public bool StartDelayed = true;
+
+    /// <summary>
+    /// Whether or not the entity should be stunned when reforming at all
+    /// </summary>
+    [DataField("shouldStun")]
+    public bool ShouldStun = true;
+
+    /// <summary>
+    /// The text that appears when attempting to reform
+    /// </summary>
+    [DataField("popupText", required: true)]
+    public string PopupText;
+
+    /// <summary>
+    /// The mob that our entity will reform into
+    /// </summary>
+    [DataField("prototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>), required: true)]
+    public string? Prototype { get; private set; }
```

Comment:
```suggestion
    /// <summary>
    /// The mob that our entity will reform into
    /// </summary>
    [DataField(required: true)]
    public EntProtoId Prototype = default!;
```
shouldnt be nullable if its required + use entprotoid + no property

Reply:
also imo rename this to like `ReformPrototype` could be confused with the action prototype id

Reply:
iirc when this wasn't nullable it caused tests to fail, but chances are using MapInit should fix that anyways.